tokio/src/util/linked_list.rs

```diff
@@ -128,6 +155,18 @@ impl<T: Link> LinkedList<T> {
         true
     }
 
+    /// Returns `true` if `node` is the first node in this list.
+    #[cfg(feature = "sync")] // used only by `batch_semaphore`
```

Comment:
No need to change this now, but for future reference. I enjoy doing separate impl blocks w/ the `cfg_if_*` macro. For example:

```rust
cfg_if_sync! {
    impl MyType {
        fn some_fn_only_needed_with_sync(&self) { ... }
    }
}